docs/advanced-js-interop.md

```diff
@@ -0,0 +1,659 @@
+# Advanced JavaScript interoperability
```

Comment:
@jchavarri Should I create more structure by making a `communicate-with-javascript` directory and putting this and the other bindings topics pages into it? Or is flat page hierarchy OK?

Reply:
Not sure. I guess it makes sense to reflect the website public hierarchy on the file system as well (i.e. create a folder). @davesnx wdyt?